app/lib/database/views/flags_view.js

```diff
@@ -12,35 +12,41 @@
  * or submit itself to any jurisdiction.
  */
 
-const flags_view = (query) => {
-    
-    rn = query.run_numbers;
-    let rn_sql = "";
-    if (typeof(rn) === 'object') {
-        rn_sql = rn.join(",");
-    } else if (typeof(rn) === 'string') {
-        rn_sql = rn
+
+const handleArray = (data, field, escape) => {
+    let sqlLogicClause = undefined;
+    if (Array.isArray(data)) {
+        sqlLogicClause = data.map((d) => escape ? `'${d}'` : d).join(',');
+    } else if (typeof(data) === 'string' || !data) {
+        sqlLogicClause = escape ? `'${data}'` : data;
     } else {
-        throw `run_numbers seems to be incorrect ${rn}`
+        throw `incorrect format <${data}> for ${field}`;
```

Comment:
Always throw errors, not string (not from me : https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw#description)

Reply:
applied